src/Bicep.Core/Text/StringExtensions.cs

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Bicep.Core.Text
+{
+    public static class StringExtensions
+    {
+        public static string RegexEscaped(this string text)
+            => text.Replace("]", @"\]")
+                .Replace("[", @"\[")
+                .Replace(@"\", @"\\")
+                .Replace("^", @"\^")
+                .Replace(".", @"\.")
+                .Replace("|", @"\|")
+                .Replace("?", @"\?")
+                .Replace("*", @"\*");
```

Comment:
Looks like there's actually a built-in .NET function for this - `Regex.Escape()`